### ISSUE TITLE
3 years since translations updated from Transifex

### ISSUE THREAD

**Issue by haaninjo** [p11-glue/p11-kit#378]
Hi,
I completed the Swedish translation of p11-kit at Transifex, and see that the last translation pull from there happened in 2018 with Commit 49d2ededb64197702a8708cb4a453497bc7eaecd. It makes me think that it is time for an update now that there has been added some more translatable strings.

**Comment by ueno** [comment]
It's a shame but I completely forgot how to pull translations from Transifex. I have the following `.tx/config`:
```
[main]
host = https://www.transifex.com

[p11-kit.p11-kit-pot]
file_filter = po/<lang>.po
minimum_perc = 0
source_file = po/p11-kit.pot
source_lang = en
type = PO
```
but I only get:
```console
$ tx pull -a
tx ERROR: Not Found
tx ERROR: Not Found
tx ERROR: Resource p11-kit.p11-kit-pot doesn't exist on the server.
tx INFO: Done.
```
Could you give me the right instruction? (cc @piotrdrag)

**Comment by piotrdrag** [comment]
Looking at https://www.transifex.com/freedesktop/p11-kit/master/, is it possible that `[p11-kit.p11-kit-pot]` should be changed to `[p11-kit.master]`?

**Comment by ueno** [comment]
Indeed, it works; thanks!

**Comment by ueno** [comment]
Sigh, I'm totally messed up: after checking `tx pull -a` worked, I cleaned up the working copy with `git checkout .` and switched to a new topic branch, then called `tx pull -a`; nothing is pulled :-( so I tried `tx pull -f`... now it put "Last-Translator: Daiki Ueno <ueno@gnu.org>\n" to all .po files, overriding the original value, which is certainly not wanted.

**Comment by rffontenelle** [comment]
I often see projects which rely on the maintainer(s) to keep translation synced, but it gets forgotten between several maintaining tasks of one or more projects. There are some ways to ease this task:

1. Add the translation sync in a list of pre-release tasks, preferable on a alpha/rc/etc. version;
2. Automate the translation sync with transifex-client tool in a CI solution like GitHub Actions
3. [Link Transifex with this repository](https://docs.transifex.com/transifex-github-integrations/github-tx-ui) for a seamless integration (see [FAQ CLI Client vs Direct Integration](https://docs.transifex.com/client/client-faq#cli-client-vs-direct-integration))

**Comment by comradekingu** [comment]
@ueno This is just one of the many reasons why Transifex is an absolute pain to use for translators, and why quality suffers when anyone uses it.
Set up shop on Weblate instead.

**Comment by haaninjo** [comment]
> Sigh, I'm totally messed up: after checking `tx pull -a` worked, I cleaned up the working copy with `git checkout .` and switched to a new topic branch, then called `tx pull -a`; nothing is pulled :-( so I tried `tx pull -f`... now it put "Last-Translator: Daiki Ueno [ueno@gnu.org](mailto:ueno@gnu.org)\n" to all .po files, overriding the original value, which is certainly not wanted.

I downloaded transifex-client now to see if I had any better luck, but got a bit surprised that it showed ueno@gnu.org as last translator for me as well for many (but not the most active) translations.

I figured out this is since Last-Translator must have been set on Transifex when the files were last pushed March 6th 2021. Languages that have had translations after this has another value for Last-Translator.

In other words, nothing strange with the pull command, the value was already changed on the Transifex server.

**Comment by comradekingu** [comment]
Borking the crediting is standard fare on Transifex.
"tx pull" only works some of the time. Devs solve this by writing scripts that run it over and over again. (Not a joke)
Considering how much of a half-a-hammer tool it is, without knowing who the drive-by translators are, it all becomes a blur without essentially having to start over again.
If someone told me Transifex was a conspiracy to ruin translations it would seem all the more convincing given how many times exactly that has happened.

**Comment by haaninjo** [comment]
> Did it also credit all string chances to you? Borking the crediting is standard fare on Transifex, so be careful. "tx pull" only works some of the time. Devs solve this by writing scripts that run it over and over again. (Not a joke).

Well, it created a single huge commit, can be seen on my fork at https://github.com/haaninjo/p11-kit/commit/73255a99a4b51f9a1881abb8fe027bc53e9ee691

But now I see other weirdness, it seems it replaces inactive transifex accounts with hashes instead? For example in the header of es.po:
```
-# Daniel Mustieles <inactive+leo@transifex.com>, 2012-2013
+# 0f759dd1ea6c4c76cedc299039ca4f23_7584d04 <454cc45058db9632179dc8079c0c1af3_5311>, 2012-2013
```

I'm more and more inclined to agree with your opinion on Transifex ;-)

**Comment by dhztao68** [comment]
Thank you for your email
